File: VIPER Tests.xctemplate/Objective-C/___FILEBASENAME___InteractorTests.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import <UIKit/UIKit.h>
#import <XCTest/XCTest.h>

#import "___FILEBASENAMEASIDENTIFIER___Interactor.h"
#import "___FILEBASENAMEASIDENTIFIER___InteractorProtocols.h"

@interface ___FILEBASENAMEASIDENTIFIER___InteractorTests : XCTestCase <___FILEBASENAMEASIDENTIFIER___InteractorOutput>

@property (nonatomic, strong) ___FILEBASENAMEASIDENTIFIER___Interactor *interactor;

// Test Objects
@property (nonatomic, strong) XCTestExpectation *expectation;

@end

@implementation ___FILEBASENAMEASIDENTIFIER___InteractorTests

- (void)setUp {
        [super setUp];
        
        self.interactor = [___FILEBASENAMEASIDENTIFIER___Interactor new];
        
        self.interactor.presenter = self;
}

- (void)tearDown {
        [super tearDown];
        self.expectation = nil;
}

#pragma mark - Operational

#pragma mark - Interactor Input

#pragma mark - Interactor Output

@end

File: VIPER Tests.xctemplate/Swift/___FILEBASENAME___InteractorTests.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import Foundation
import XCTest

@testable import ___PROJECTNAME___

class ___FILEBASENAMEASIDENTIFIER___InteractorTests: XCTestCase {
        var interactor = ___FILEBASENAMEASIDENTIFIER___Interactor()

        // MARK: - Test Objects
        var expectation: XCTestExpectation?

        override func setUp() {
                super.setUp()

                interactor = ___FILEBASENAMEASIDENTIFIER___Interactor()

                interactor.presenter = self
        }

        override func tearDown() {
                super.tearDown()
                expectation = nil
        }

        // MARK: - Operational

        // MARK: - Presenter to Interactor Interface

}

// MARK: - Interactor to Presenter Interface
extension ___FILEBASENAMEASIDENTIFIER___InteractorTests: ___FILEBASENAMEASIDENTIFIER___InteractorToPresenterInterface {

}
